test/unit/entities/app-installation-test.js

```diff
@@ -0,0 +1,65 @@
+import test from 'blue-tape'
+import {cloneMock} from '../mocks/entities'
+import setupHttpMock from '../mocks/http'
+import {wrapAppInstallation, wrapAppInstallationCollection} from '../../../lib/entities/app-installation'
+
+import {
+  entityWrappedTest,
+  entityCollectionWrappedTest,
+  failingVersionActionTest,
+  entityDeleteTest,
+  failingActionTest
+} from '../test-creators/instance-entity-methods'
+
+function setup (promise) {
+  return {
+    httpMock: setupHttpMock(promise),
+    entityMock: cloneMock('appInstallation')
+  }
+}
+
+test('AppInstallation is wrapped', (t) => {
+  entityWrappedTest(t, setup, {
+    wrapperMethod: wrapAppInstallation
+  })
+})
+
+test('AppInstallation collection is wrapped', (t) => {
+  return entityCollectionWrappedTest(t, setup, {
+    wrapperMethod: wrapAppInstallationCollection
+  })
+})
+
+test('AppInstallation update', (t) => {
+  t.plan(2)
+  const { httpMock, entityMock } = setup()
+  entityMock.sys.version = 2
+  const entity = wrapAppInstallation(httpMock, entityMock)
+  entity.name = 'updatedname'
+  return entity.update()
+    .then((response) => {
+      t.ok(response.toPlainObject, 'response is wrapped')
+      t.equals(httpMock.put.args[0][1].name, 'updatedname', 'data is sent')
+      return {httpMock, entityMock, response}
+    })
+})
+
+test('AppInstallation update fails', (t) => {
+  return failingVersionActionTest(t, setup, {
```

Comment:
I think there's a better "failingActionTest" here that's better than failingVersionActionTest (especially if appInstallation isn't minding version numbers in the same way? did I understand that right?)

Reply:
@FLoppix